src/app/courses/course/course.component.ts

```diff
@@ -15,16 +16,21 @@ export class CourseComponent implements OnInit {
   constructor(private route: ActivatedRoute, private coursesService: CoursesService, private router: Router) { }
 
   ngOnInit(): void {
-    this.course = this.coursesService.getItemById(+this.route.snapshot.paramMap.get('id'));
+    const _this = this;
+    this.coursesService.getItemById(+this.route.snapshot.paramMap.get('id'))
+      .subscribe(
+        course => { _this.course = course; },
+        error => { console.log(error); }
+      );
```

Comment:
1. Please, keep subscription reference to unsubscribe it in `ngOnInit`
2. Keep `subscribe` block empty. Use `tap` operator to handle side-effects that don't influence on initial stream and `catchError` for handling errors.